packages/react-front-kit/src/hooks/useStorybookArgsConnect.ts

```diff
@@ -0,0 +1,30 @@
+import { useArgs } from '@storybook/preview-api';
+
+import { isCallback, removeNulls } from '../utils/utilities';
+
+export function useStorybookArgsConnect<T extends Record<string, unknown>>(
+  args: T,
+  propsToConnect: Record<string, string>
+): T {
+  const [, setArgs] = useArgs<T>();
+  const connectedProps = Object.entries(propsToConnect)
+    .map(([key, value]) => {
+      const callbackProp: ((e: unknown) => void) | null = isCallback(args[key])
+        ? (args[key] as (e: unknown) => void)
+        : null;
+      return callbackProp !== null
+        ? {
+            [key]: (v: unknown): void => {
+              if (isCallback(callbackProp)) {
+                callbackProp?.(v);
+                if (args[value] !== undefined) {
+                  setArgs({ [value]: v } as Partial<T>);
+                }
+              }
+            },
+          }
+        : null;
+    })
+    .filter(removeNulls);
+  return { ...args, ...Object.assign({}, ...connectedProps) };
```

Comment:
`.Object.assign` n'apporte rien ici:
```suggestion
  return { ...args, ...connectedProps };
```

Reply:
Il est nécessaire, connectedProps est un array, le spread operator ici donne `{ 0: {key: value}, 1: {key: value}}` au lieu du `{ key:value, key:value }`
Object.assign( {}, ...} permet de spread l'array correctement.

Reply:
@tonai Cette syntax ne passe pas, le typage ne correspond pas. Un très long trace typescript qui inclut par exemple `Type  { [x: string]: (v: unknown) => void; }  is not assignable to type  readonly [PropertyKey, (v: unknown) => void]` et 22 autre erreurs de typages.